packages/angular/projects/ui-angular/src/lib/services/state-machine.service.ts

```diff
@@ -19,15 +20,21 @@ export class StateMachineService {
   private _authState: AuthMachineState;
   private _authService: AuthInterpreter;
   private _user: Record<string, any>; // TODO: strongly type CognitoUser
+  private _services: ReturnType<typeof getSendAliases>;
 
-  public get services() {
-    return {
-      submit: (formData) =>
-        this._authService.send({ type: 'SUBMIT', data: formData }),
-    } as const;
+  constructor() {
+    const interpreter = interpret(authMachine, { devTools: true })
+      .onTransition((state) => {
+        this._user = state.context.user;
+        this._authState = state;
+      })
+      .start();
+    this._services = getSendAliases(interpreter.send);
+    this._authService = interpreter;
```

Comment:
I added angular integration with this new helper function for demonstration purposes. Angular unfortunately does not have dedicated xstate package, so it's a little bit more messy than what would be on React/Vue.